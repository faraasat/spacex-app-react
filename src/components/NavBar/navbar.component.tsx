import "./navbar.styles.css";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import { Link } from "react-router-dom";
import IconButton from "@material-ui/core/IconButton";
import { Typography } from "@material-ui/core";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    menuButton: {
      marginRight: theme.spacing(2),
    },
  })
);

export default function NavBar() {
  const classes = useStyles();

  return (
    <AppBar position="static" style={{ backgroundColor: "rgba(1, 1, 1, 0.562)" }}>
      <Toolbar>
        <IconButton
          edge="start"
          className={classes.menuButton}
          color="inherit"
          aria-label="menu"
        >
          SpaceX App
        </IconButton>
        <Link to="/" style={{ textDecoration: "none" }}>
          <Typography style={{ paddingLeft: 20, color: "white" }} variant="h6">
            Home
          </Typography>
        </Link>
        <Link to="/explore" style={{ textDecoration: "none" }}>
          <Typography style={{ paddingLeft: 20, color: "white" }} variant="h6">
            Explore
          </Typography>
        </Link>
      </Toolbar>
    </AppBar>
  );
}
