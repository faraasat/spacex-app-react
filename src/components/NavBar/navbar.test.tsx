import { shallow } from "enzyme";
import NavBar from "./navbar.component";

describe("Home Component", () => {
  const shallowRender = shallow(<NavBar />);

  it("should match snapshot", () => {
    expect(shallowRender.debug()).toMatchSnapshot();
  });

  it("should find 0 or more on div", () => {
    expect(shallowRender.find("div").length).toBeGreaterThanOrEqual(0);
  });
});
