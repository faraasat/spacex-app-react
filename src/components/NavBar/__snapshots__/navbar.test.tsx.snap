// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should match snapshot 1`] = `
"<WithStyles(ForwardRef(AppBar)) position=\\"static\\" style={{...}}>
  <WithStyles(ForwardRef(Toolbar))>
    <WithStyles(ForwardRef(IconButton)) edge=\\"start\\" className=\\"makeStyles-menuButton-1\\" color=\\"inherit\\" aria-label=\\"menu\\">
      SpaceX App
    </WithStyles(ForwardRef(IconButton))>
    <Link to=\\"/\\" style={{...}}>
      <WithStyles(ForwardRef(Typography)) style={{...}} variant=\\"h6\\">
        Home
      </WithStyles(ForwardRef(Typography))>
    </Link>
    <Link to=\\"/explore\\" style={{...}}>
      <WithStyles(ForwardRef(Typography)) style={{...}} variant=\\"h6\\">
        Explore
      </WithStyles(ForwardRef(Typography))>
    </Link>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>"
`;
